File: Src/Engine/Editor/SceneGraph/Actor/SceneGraphNode.cpp

#include "SceneGraphNode.h"

#include "Editor/GUI/Tree/ActorTreeNode.h"
#include "Runtime/Level/Scene/Scene.h"

namespace SE::Editor
{
	SceneGraphNode::SceneGraphNode(Scene* scene) : ActorGraphNode(scene)
	{
	}

	void SceneGraphNode::__SetIsEdited(bool value)
	{
		if (_isEdited != value)
		{
			_isEdited = value;
			_treeNode->UpdateText();
		}
	}
	Scene* SceneGraphNode::__GetScene()
	{
		return (Scene*)_actor;
	}
} // SE

File: Src/Engine/Editor/SceneGraph/SceneGraphNode.cs
using System;

namespace SE.Editor.SceneGraph
{
    /// <summary>
    /// Scene graph node for a loaded runtime scene.
    /// </summary>
    public sealed class SceneGraphNode : ActorGraphNode
    {
        private bool m_IsEdited;

        public SceneGraphNode(SceneGraphFactory factory, SE.Scene scene)
            : base(factory, scene)
        {
        }

        public bool IsEdited
        {
            get => m_IsEdited;
            set
            {
                if (m_IsEdited == value)
                    return;

                m_IsEdited = value;
                TreeNode.UpdateText();
            }
        }

        public SE.Scene Scene => (SE.Scene)Actor!;

        public override SceneGraphNode ParentScene => this;
    }
}
